Exercise_3.py

```diff
@@ -17,16 +20,51 @@ def append(self, data):
         Insert a new element at the end of the list.
         Takes O(n) time.
         """
-        
+        new = ListNode(data)
+        if self.head:                           #checks if there's a head, if true then it traverses at the end and adds the new node. Else, makes the new node head 
+            while self.head.next:
+                self.head = self.head.next
+            self.head.next = new
+        else:
+            self.head = new        
+
     def find(self, key):
         """
         Search for the first element with `data` matching
         `key`. Return the element or `None` if not found.
         Takes O(n) time.
         """
-        
+        tmp = self.head                        # to create an instance
+        while tmp:                       # traverses the list while there is a head.
+            if tmp.data == key:
+                return key
+            else:
+                tmp = tmp.next
+        return None            
     def remove(self, key):
         """
         Remove the first occurrence of `key` in the list.
         Takes O(n) time.
         """
+        tmp = self.head
+        if tmp.data == key:
+            self.head = self.head.next
+        elif tmp.next:
+            while tmp.next:                       
+                if tmp.next.data == key:
+                    tmp.next = tmp.next.next
+                    #return key
+                    break
+                tmp = tmp.next
+        else:
+            return "Not in the linkedList"  
+
```

Comment:
You can simply add this statement print(f"The element removed from the list is:  {key}") and it will print the removed key. Is this what you meant by "not able to return the key?" You have already achieved your objective of removing the first occurrence of the key in the list.

Reply:
I tried to return the key when I was removing it, but it gives NoneType object. I got your point, that I should print the key instead and break it.